src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -1011,16 +1011,21 @@ class PosePriorBundleAdjuster : public BundleAdjuster {
 
   bool AlignReconstruction() {
     RANSACOptions ransac_options;
-    ransac_options.max_error = -1;
-    size_t num_covs = 0;
-    Eigen::Vector3d avg_cov = Eigen::Vector3d::Zero();
-    for (const auto& [_, pose_prior] : pose_priors_) {
-      if (pose_prior.IsCovarianceValid()) {
-        avg_cov += pose_prior.position_covariance.diagonal();
-        ++num_covs;
+    if (prior_options_.ransac_max_error > 0) {
+      ransac_options.max_error = prior_options_.ransac_max_error;
+    } else {
+      size_t num_covs = 0;
+      Eigen::Vector3d avg_cov = Eigen::Vector3d::Zero();
+      for (const auto& [_, pose_prior] : pose_priors_) {
+        if (pose_prior.IsCovarianceValid()) {
+          avg_cov += pose_prior.position_covariance.diagonal();
+          ++num_covs;
+        }
+      }
+      if (num_covs == 0) {
+        LOG(WARNING) << "No pose priors with valid covariance found.";
+        return false;
       }
-    }
-    if (num_covs > 0) {
       ransac_options.max_error = (3. * (avg_cov / num_covs).cwiseSqrt()).norm();
```

Comment:
Niche and not directly relevant to this PR: cWiseSqrt() + norm() should be numerically inferior to the square root of the sum?

Reply:
Not sure I understand. sqrt(sum(...)) would not produce the same result?

Reply:
vec.cwiseSqrt().norm() and sqrt(vec.sum()) are mathematically equivalent?

Reply:
Actually, looking at this more now, I would have expected the multiplier 3 to be outside the norm? I guess it is meant as the confidence interval, where we set the max error as 3 * stddev?

Reply:
I am fully confused now. That's also the same. Excuse the brain fart :-)